### DISCUSSION TITLE
Isotropic remeshing returns faces referencing unexisting vertices

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#6625] [Q&A] [answered]
Hello,

  When I run [this code](https://gist.github.com/stla/61a9b6e1ca9b65a018f572ab70034413) (or rather the analogous code in R), then I get a mesh with 2550 vertices but the face indices of this mesh take the values `0->2561` instead of `0->2549`. If I remove the faces referencing a vertex in `0->11` and I subtract `12` to the face indices (so the face indices take values `0->2549`), then the mesh I get is correct. Note that `12` is the number of vertices of the original mesh.

  I already encountered a similar issue when I did some successive intersections of meshes using the same output mesh: there was some kind of cumulation instead of a replacement. I think the above C++ code is correct, this is an issue with R (or with my R/C++ code). Would you have a guess of what happens with my R/C++ implementation?

**Comment by sloriot** [comment, marked as answer]
Are you aware of the use of garbage in Surface_mesh? See [here](https://doc.cgal.org/latest/Surface_mesh/index.html#sectionSurfaceMesh_memory)

**Reply by stla** [reply]
Thanks. I knew this function, I use it to remove degenerate faces. But I didn't know I had to use it here. I'm going to try now.

**Reply by stla** [reply]
Yeah, this works!! Thanks!

**Reply by afabri** [reply]
@sloriot should we add a precondition that there is no garbage, or is it more complex than that (for example in case the users provides an index map)?  Is there a warning/note to add in the documentation (here as well as elsewhere)?

**Reply by sloriot** [reply]
it is documented that indices are in the range `[0 -> num_vertices(tm)[`

**Comment by stla** [comment]
Also needed for `CGAL::Polygon_mesh_processing::clip`.
